src/project-management/android-app.ts

```diff
@@ -14,146 +14,108 @@
  * limitations under the License.
  */
 
-import { FirebaseProjectManagementError } from '../utils/error';
-import * as validator from '../utils/validator';
-import { ProjectManagementRequestHandler, assertServerResponse } from './project-management-api-request';
-import { AndroidAppMetadata, AppPlatform } from './app-metadata';
+import { AndroidAppMetadata } from './app-metadata';
 
-export class AndroidApp {
-  private readonly resourceName: string;
-
-  constructor(
-      public readonly appId: string,
-      private readonly requestHandler: ProjectManagementRequestHandler) {
-    if (!validator.isNonEmptyString(appId)) {
-      throw new FirebaseProjectManagementError(
-        'invalid-argument', 'appId must be a non-empty string.');
-    }
-
-    this.resourceName = `projects/-/androidApps/${appId}`;
-  }
-
-  public getMetadata(): Promise<AndroidAppMetadata> {
-    return this.requestHandler.getResource(this.resourceName)
-      .then((responseData: any) => {
-        assertServerResponse(
-          validator.isNonNullObject(responseData),
-          responseData,
-          'getMetadata()\'s responseData must be a non-null object.');
-
-        const requiredFieldsList = ['name', 'appId', 'projectId', 'packageName'];
-        requiredFieldsList.forEach((requiredField) => {
-          assertServerResponse(
-            validator.isNonEmptyString(responseData[requiredField]),
-            responseData,
-            `getMetadata()'s responseData.${requiredField} must be a non-empty string.`);
-        });
-
-        const metadata: AndroidAppMetadata = {
-          platform: AppPlatform.ANDROID,
-          resourceName: responseData.name,
-          appId: responseData.appId,
-          displayName: responseData.displayName || null,
-          projectId: responseData.projectId,
-          packageName: responseData.packageName,
-        };
-        return metadata;
-      });
-  }
-
-  public setDisplayName(newDisplayName: string): Promise<void> {
-    return this.requestHandler.setDisplayName(this.resourceName, newDisplayName);
-  }
-
-  public getShaCertificates(): Promise<ShaCertificate[]> {
-    return this.requestHandler.getAndroidShaCertificates(this.resourceName)
-      .then((responseData: any) => {
-        assertServerResponse(
-          validator.isNonNullObject(responseData),
-          responseData,
-          'getShaCertificates()\'s responseData must be a non-null object.');
-
-        if (!responseData.certificates) {
-          return [];
-        }
-
-        assertServerResponse(
-          validator.isArray(responseData.certificates),
-          responseData,
-          '"certificates" field must be present in the getShaCertificates() response data.');
-
-        const requiredFieldsList = ['name', 'shaHash'];
+/**
+ * A SHA-1 or SHA-256 certificate.
+ *
+ * Do not call this constructor directly. Instead, use
+ * [`projectManagement.shaCertificate()`](admin.projectManagement.ProjectManagement#shaCertificate).
```

Comment:
Unsure if the documentation links are still functional, I will need to look into this more carefully.

Reply:
This might be ok for now, but based on my initial testing the document generation script will need some editing to properly generate docs (look into `lib` for typings, recursively search directories, and then flatten the typing directories accordingly).